### DISCUSSION TITLE
why method foreach() or stream() for ElementCollections are deprecated ?

### DISCUSSION THREAD

**Discussion by Tonicku** [selenide/selenide#1699] [Q&A] [answered]
Usage code:
SERIES_CHECKBOXES.forEach(SelenideElement::click);
SERIES_CHECKBOXES.stream()
                                    .noneMatch(e -> e.is(selected))

where SERIES_CHECKBOXES is a ElementsCollection

What the reason to get these methods and few others from the ElementsCollection class have been deprecated ?

**Comment by asolntsev** [comment, marked as answer]
@Tonicku Hi. 
Did you read the release notes?

https://selenide.org/2022/01/10/selenide-6.2.0/ -> "Replaced $$.iterator() by $$.asDynamicIterable() and $$.asFixedIterable()"

**Reply by Tonicku** [reply]
ok i gonna read it :)
